apple/RNGestureHandlerModule.mm

```diff
@@ -203,6 +203,38 @@ - (void)setBridge:(RCTBridge *)bridge
 #endif // RCT_NEW_ARCH_ENABLED
 }
 
+#if TARGET_OS_OSX
+- (void)setGestureState:(int)state forHandler:(int)handlerTag
+{
+  RNGestureHandler *handler = [_manager handlerWithTag:@(handlerTag)];
```

Comment:
Wouldn't something like this work?

```objective-c
#if TARGET_OS_OSX
    NSGestureRecognizerState failed = NSGestureRecognizerStateFailed;
    NSGestureRecognizerState began = NSGestureRecognizerStateBegan;
    NSGestureRecognizerState cancelled = NSGestureRecognizerStateCancelled;
    NSGestureRecognizerState ended = NSGestureRecognizerStateEnded;
    NSGestureRecognizerState possible = NSGestureRecognizerStatePossible;
#else
    UIGestureRecognizerState failed = UIGestureRecognizerStateFailed;
    UIGestureRecognizerState began = UIGestureRecognizerStateBegan;
    UIGestureRecognizerState cancelled = UIGestureRecognizerStateCancelled;
    UIGestureRecognizerState ended = UIGestureRecognizerStateEnded;
    UIGestureRecognizerState possible = UIGestureRecognizerStatePossible;
#endif
    
  if (handler != nil) {
    if (state == 1) { // FAILED
      handler.recognizer.state = failed;
    } else if (state == 2) { // BEGAN
      handler.recognizer.state = possible;
    } else if (state == 3) { // CANCELLED
      handler.recognizer.state = cancelled;
    } else if (state == 4) { // ACTIVE
      [handler stopActivationBlocker];
      handler.recognizer.state = began;
    } else if (state == 5) { // ENDED
      handler.recognizer.state = ended;
    }
  }
```

Reply:
It should work ok with macOS, however I haven't been able to test that on iOS yet, since I'm getting some weird errors with `MaskedView` and `SafeAreaProvider`

Reply:
Another solution would be to use macros like we do in Reanimated, e.g. `REAView` represents `UIView`/`NSView`:
```objc++
#if TARGET_OS_OSX
#define RNGHGestureRecognizerStateFailed NSGestureRecognizerStateFailed;
#else
#define RNGHGestureRecognizerStateFailed UIGestureRecognizerStateFailed;
#endif
```
However, the recommended way to achieve it in C++ is to use `constexpr`:
```objc++
#if TARGET_OS_OSX
constexpr auto RNGHGestureRecognizerStateFailed = NSGestureRecognizerStateFailed;
#else
constexpr auto RNGHGestureRecognizerStateFailed = UIGestureRecognizerStateFailed;
#endif
```

Reply:
This one seems like the best option. Maybe we could do the same thing `CGPoint`/`NSPoint` etc. - this way we could avoid a lot of duplications.

Reply:
I used defines, as `constexpr` caused some errors. Is there any reason why it's recommended approach?